### PR TITLE
Improve TypeScript recipe for ESM and ts-node usage

### PR DESCRIPTION
**Pull request by mesqueeb**

### CODE REVIEW

**Comment by sindresorhus** [comment]
I pushed some updates. Please don't force push over them.

File: docs/recipes/typescript.md [hunk 1]
```diff
@@ -8,13 +8,74 @@ This guide assumes you've already set up TypeScript for your project. Note that
...
+	"ava": {
+		"extensions": {
+			"ts": "module"
+		},
+		"nonSemVerExperiments": {
+			"configurableModuleFormat": true
+		},
+		"nodeArguments": [
+			"--loader",
+			"ts-node/esm"
```
Comment (novemberborn) [inline docs/recipes/typescript.md:43]: Would this work?

```suggestion
			"--loader=ts-node/esm"
```

File: docs/recipes/typescript.md [hunk 2]
```diff
@@ -8,13 +8,74 @@ This guide assumes you've already set up TypeScript for your project. Note that
...
+	}
+}
+```
+
+You also need to have this in your `tsconfig.json`:
+
+```json
+{
+	"compilerOptions": {
+		"module": "ESNext",
```
Comment (novemberborn) [inline docs/recipes/typescript.md:54]: Should this be `ES2020`?

```suggestion
		"module": "ES2020",
```

**Comment by novemberborn** [comment]
Thanks for doing this @mesqueeb!

File: docs/recipes/typescript.md (hunk 1 above)
Comment (mesqueeb) [reply docs/recipes/typescript.md:43]: I can try!

File: docs/recipes/typescript.md (hunk 2 above)
Comment (mesqueeb) [reply docs/recipes/typescript.md:54]: I will try with 2020 and let 5ou know !

File: docs/recipes/typescript.md (hunk 1 above)
Comment (mesqueeb) [reply docs/recipes/typescript.md:43]: @novemberborn yes this worked
image

File: docs/recipes/typescript.md (hunk 2 above)
Comment (mesqueeb) [reply docs/recipes/typescript.md:54]: @novemberborn this also worked
image

**Comment by mesqueeb** [comment]
@novemberborn thanks! tested & updated your suggestions, ready for review

**Comment by novemberborn** [comment]
Thanks @mesqueeb!

**Pull request merged**